press/press/doctype/press_role/test_press_role.py

```diff
@@ -0,0 +1,163 @@
+		self.team_user = create_user("team@example.com")
+		self.team = create_test_team(self.team_user.email)
+		self.team_member = create_user("user123@example.com")
+		self.team.append("team_members", {"user": self.team_member.name})
+		self.team.save()
+		self.perm_role = create_permission_role(self.team.name)
+		self.perm_role2 = create_permission_role(self.team.name)
+
+	def tearDown(self):
+		frappe.set_user("Administrator")
+		frappe.delete_doc("Press Role", self.perm_role.name, force=True)
+		frappe.delete_doc("Press Role", self.perm_role2.name, force=True)
+		frappe.delete_doc("Team", self.team.name, force=True)
+		frappe.delete_doc("User", self.team_member.name, force=True)
+		frappe.delete_doc("User", self.team_user.name, force=True)
+		frappe.local._current_team = None
+
+	def test_add_user(self):
+		self.perm_role.add_user(self.team_member.name)
+		perm_role_users = get_users(self.perm_role)
+		perm_role_user_exists = any(
+			self.team_member.name == perm_role_user.user for perm_role_user in perm_role_users
+		)
+		self.assertTrue(perm_role_user_exists)
+		self.assertRaises(
+			frappe.ValidationError, self.perm_role.add_user, self.team_member.name
+		)
+
+	def test_remove_user(self):
+		self.perm_role.add_user(self.team_member.name)
+		self.perm_role.remove_user(self.team_member.name)
+		perm_role_users = get_users(self.perm_role)
+		perm_role_user_exists = any(
+			self.team_member.name == perm_role_user.user for perm_role_user in perm_role_users
+		)
+		self.assertFalse(perm_role_user_exists)
+		self.assertRaises(
+			frappe.ValidationError, self.perm_role.remove_user, self.team_member.name
+		)
+
+	def test_delete_role(self):
+		perm = frappe.new_doc("Press Role Permission")
+		perm.role = self.perm_role.name
+		perm.team = self.team.name
+		perm.save()
+
+		self.perm_role.delete()
+		self.assertFalse(frappe.db.exists("Press Role", self.perm_role.name))
+		self.assertFalse(
+			frappe.db.get_all("Press Role Permission", filters={"role": self.perm_role.name})
+		)
+
+	def test_delete_permissions(self):
+		perm = frappe.new_doc("Press Role Permission")
+		perm.role = self.perm_role.name
+		perm.team = self.team.name
+		perm.save()
+
+		permissions = frappe.get_all(
+			"Press Role Permission", filters={"role": self.perm_role.name}, pluck="name"
+		)
+		self.perm_role.delete_permissions(permissions)
+		self.assertFalse(
+			frappe.db.get_all("Press Role Permission", filters={"role": self.perm_role.name})
+		)
+
+	def test_get_list_with_permissions(self):
+		from press.api.client import get_list
+
+		frappe.set_user("Administrator")
+		site1 = create_test_site(team=self.team.name)
+		site2 = create_test_site(team=self.team.name)
+		self.perm_role.add_user(self.team_user.name)
+		self.perm_role2.add_user(self.team_user.name)
+		frappe.set_user(self.team_user.name)
+
+		# no permissions added should show all records
+		self.assertCountEqual(get_list("Site"), [{"name": site1.name}, {"name": site2.name}])
+		return
```

Comment:
unintended return?

Reply:
Oops, yes unintended.  Was trying to run that particular case only 😅 .